head.asm

```diff
@@ -43,7 +43,19 @@ load_gdt:
 	ret
 
 keyboard_handler:                 
+	push ds
+	push es
+	push fs
+	push gs
+	pushad
+
 	call    keyboard_handler_main
+
+	popad
+	pop ds
```

Comment:
I'd say you would be right. I'm guessing I never noticed this bug because all of those segments probably contain the same value everywhere in the program. but obviously it's still a bug and it would easily blow things up if/when features are added that make use of the segment registers (Such as using them for CPU-local data).

On that note it also appears the segments are also never assigned, so it's quite likely this code got lucky in other areas. A while back I had an issue stemming from something similar in my kernel, it's fixed [here](https://github.com/DSMan195276/protura/blob/master/arch/x86/kernel/cpu.c#L42). The code in this PR doesn't reload the segments after the `lgdt`, but that's necessary to ensure they actually contain the correct values and also to ensure they start utilizing the newly defined segments.